runtime/src/bank_forks.rs

```diff
@@ -163,9 +174,20 @@ impl BankForks {
         for parent in bank.proper_ancestors() {
             self.descendants.entry(parent).or_default().insert(slot);
         }
+        if self.should_be_in_vote_only_mode() && !self.in_vote_only_mode.load(Ordering::Relaxed) {
+            datapoint_info!(
+                "bank_forks-entering-vote-only-mode",
+                ("banks_len", self.banks.len(), i64),
+            );
+            self.in_vote_only_mode.store(true, Ordering::Relaxed);
+        }
         bank
     }
 
+    fn should_be_in_vote_only_mode(&self) -> bool {
+        self.highest_slot().saturating_sub(self.root()) >= MAX_ROOT_DISTANCE_FOR_VOTE_ONLY
```

Comment:
Using the highest slot seems like an attack vector if someone can generate a bank that's continually 400 slots ahead of the root

I think using the `heaviest_bank` here might be a bit safer: https://github.com/solana-labs/solana/blob/master/core/src/replay_stage.rs#L606 because it's a bit harder to consistently make the heaviest bank if other earlier leaders are generating blocks as well